app/importer.js

```diff
@@ -0,0 +1,151 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'strict mode'
+
+const electron = require('electron')
+const importer = electron.importer
+const dialog = electron.dialog
+const BrowserWindow = electron.BrowserWindow
+const Immutable = require('immutable')
+const siteUtil = require('../js/state/siteUtil')
+const AppStore = require('../js/stores/appStore')
+const siteTags = require('../js/constants/siteTags')
+const appActions = require('../js/actions/appActions')
+const messages = require('../js/constants/messages')
+const Filtering = require('./filtering')
+
+exports.init = () => {
+  importer.initialize()
+}
+
+exports.importData = (selected) => {
+  if (selected !== undefined) {
+    importer.importData(selected.toJS())
+  }
+}
+
+exports.importHTML = () => {
+  const files = dialog.showOpenDialog({
+    properties: ['openFile'],
+    filters: [{
+      name: 'HTML',
+      extensions: ['html', 'htm']
+    }]
+  })
+  if (files && files.length > 0) {
+    const file = files[0]
+    importer.importHTML(file)
+  }
+}
+
+importer.on('update-supported-browsers', (e, detail) => {
+  if (BrowserWindow.getFocusedWindow()) {
+    BrowserWindow.getFocusedWindow().webContents.send(messages.IMPORTER_LIST, detail)
+  }
+})
+
+importer.on('show-warning-dialog', (e) => {
+})
+
+importer.on('add-password-form', (e, detail) => {
+})
+
+importer.on('add-history-page', (e, history, visitSource) => {
+  let sites = []
+  for (let i = 0; i < history.length; ++i) {
+    const site = {
+      title: history[i].title,
+      location: history[i].url,
+      lastAccessedTime: history[i].last_visit * 1000
+    }
+    sites.push(site)
+  }
+  appActions.addSite(Immutable.fromJS(sites))
+})
+
+importer.on('add-homepage', (e, detail) => {
+})
+
+importer.on('add-bookmarks', (e, bookmarks, topLevelFolder) => {
```

Comment:
this handler has a LOT going on in it... Any chance we could break this (or parts of it) out into a method? Looks generic enough (not dependent on electron) to put into a utils class, which means we could eventually unit test it too :smile: